runtime/ftplugin/lua.vim

```diff
@@ -48,7 +48,101 @@ if (has("gui_win32") || has("gui_gtk")) && !exists("b:browsefilter")
   let b:undo_ftplugin ..= " | unlet! b:browsefilter"
 endif
 
+" fold Lua functions
+silent! function FoldLuaLevel(lnum)
```

Comment:
why the silent? Also wouldn't the following make more sense:
```vim
function FoldLuaLevel(lnum)
  if !has('lua')
     return 0
  else
     lua foldlevel_luacode()
  endif
endfunc
```

Reply:
Yes, I completely rewrote the function into Vimscript for compatibility. For performance, maybe Vim9script is sensible. Also, this can supposedly be done more efficiently by Syntax folding. I tried to use `b:changedtick` to avoid recomputes, but somehow this didn't work, likely since the folding function is executed quite early, before the whole buffer is read?

Reply:
This however, just solves specifically what [fastfold](https://github.com/Konfekt/FastFold) solves generically